### PR TITLE
Make ingestion resilient

### PR DESCRIPTION
**Pull request by karlbaker02**
This PR adds error handling for updates to the Publishing API to deal with `GdsApi::HTTPBadGateway`, `GdsApi::HTTPUnavailable`, `GdsApi::HTTPGatewayTimeout` and `GdsApi::TimedOutException` exceptions.

Previously any of these exceptions would have caused the pipeline to fail, preventing full link ingestion.

This PR also fixes an existing test to wait between batches by delegating `Kernel.sleep` to an instance method to allow for testing.

### CODE REVIEW

File: lib/related_links_updater.rb [hunk 1]
```diff
@@ -32,9 +33,9 @@ def update_related_links
 
       puts "Finished batch ##{batch_index}"
 
-      if group.length == updates_per_batch
+      if batch_index < last_group_index
         puts "Waiting for 20 minutes before starting next batch..."
```
Comment (oscarwyatt) [inline lib/related_links_updater.rb]: Really pedantic but you've got a hard coded time and a variable here, I'd suggest calculating the time from the variable

File: lib/related_links_updater.rb [hunk 2]
```diff
@@ -54,17 +55,31 @@ def update_related_links
...
+        add_to_failed_updates(source_content_id, "Failed with status #{response.code}")
+      end
+
+    rescue GdsApi::HTTPBadGateway, GdsApi::HTTPUnavailable, GdsApi::HTTPGatewayTimeout, GdsApi::TimedOutException => e
+      add_to_failed_updates(source_content_id, e)
     end
+
+  end
+
+  def wait(seconds)
```
Comment (oscarwyatt) [inline lib/related_links_updater.rb]: More out of interest than anything else, what's the reason for the abstraction here? If it's a one liner which is effectively just an alias I'm not especially in favour of it. If there is a good reason for it, maybe stick in a comment explaining why

File: lib/related_links_updater.rb (hunk 1 above)
Comment (karlbaker02) [reply lib/related_links_updater.rb]: That's fair enough 🙂

File: lib/related_links_updater.rb (hunk 2 above)
Comment (karlbaker02) [reply lib/related_links_updater.rb]: See the commit message for my first commit in this PR.

**Pull request merged**